File: CS/C#/MyLayoutControl/MyLayoutControlGroupPainter.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraLayout.Registrator;
using DevExpress.XtraLayout;
using DevExpress.LookAndFeel;
using DevExpress.XtraLayout.Painting;
using DevExpress.XtraLayout.ViewInfo;
using DevExpress.Utils.Drawing;
using DevExpress.Skins;

namespace WindowsApplication1
{
    public class MyLayoutControlGroupPainter : LayoutControlGroupPainter
    {

        private MyLayoutControl _LayoutControl;
        public MyLayoutControlGroupPainter(MyLayoutControl lc)
        {
            _LayoutControl = lc;
        }


        protected override void DrawBackground(BaseLayoutItemViewInfo e)
        {
            CustomDrawEventArgs args = new CustomDrawEventArgs(e);
            base.DrawBackground(e);
            _LayoutControl.OnCustomDrawBackground(args);
            if (args.Handled)
                return;
            base.DrawBackground(e);
        }

        protected override void DrawSelection(BaseLayoutItemViewInfo e)
        {
            CustomDrawEventArgs args = new CustomDrawEventArgs(e);
            base.DrawSelection(e);
            _LayoutControl.OnCustomDrawSelection(args);
            if (args.Handled)
                return;
            base.DrawSelection(e);
        }

        protected override void DrawTextArea(BaseLayoutItemViewInfo e)
        {
            CustomDrawEventArgs args = new CustomDrawEventArgs(e);
            base.DrawTextArea(e);
            _LayoutControl.OnCustomDrawTextArea(args);
            if (args.Handled)
                return;
            base.DrawTextArea(e);
        }

        public override void DrawControlsArea(BaseLayoutItemViewInfo e)
        {
            CustomDrawEventArgs args = new CustomDrawEventArgs(e);
            base.DrawControlsArea(e);
            _LayoutControl.OnCustomDrawControlsArea(args);
            if (args.Handled)
                return;
            base.DrawControlsArea(e);
        }
    }
}

File: VB/VB/MyLayoutControl/MyLayoutControlGroupPainter.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraLayout.Registrator
Imports DevExpress.XtraLayout
Imports DevExpress.LookAndFeel
Imports DevExpress.XtraLayout.Painting
Imports DevExpress.XtraLayout.ViewInfo
Imports DevExpress.Utils.Drawing
Imports DevExpress.Skins

Namespace WindowsApplication1
	Public Class MyLayoutControlGroupPainter
		Inherits LayoutControlGroupPainter

		Private _LayoutControl As MyLayoutControl
		Public Sub New(ByVal lc As MyLayoutControl)
			_LayoutControl = lc
		End Sub


		Protected Overrides Sub DrawBackground(ByVal e As BaseLayoutItemViewInfo)
			Dim args As New CustomDrawEventArgs(e)
			MyBase.DrawBackground(e)
			_LayoutControl.OnCustomDrawBackground(args)
			If args.Handled Then
				Return
			End If
			MyBase.DrawBackground(e)
		End Sub

		Protected Overrides Sub DrawSelection(ByVal e As BaseLayoutItemViewInfo)
			Dim args As New CustomDrawEventArgs(e)
			MyBase.DrawSelection(e)
			_LayoutControl.OnCustomDrawSelection(args)
			If args.Handled Then
				Return
			End If
			MyBase.DrawSelection(e)
		End Sub

		Protected Overrides Sub DrawTextArea(ByVal e As BaseLayoutItemViewInfo)
			Dim args As New CustomDrawEventArgs(e)
			MyBase.DrawTextArea(e)
			_LayoutControl.OnCustomDrawTextArea(args)
			If args.Handled Then
				Return
			End If
			MyBase.DrawTextArea(e)
		End Sub

		Public Overrides Sub DrawControlsArea(ByVal e As BaseLayoutItemViewInfo)
			Dim args As New CustomDrawEventArgs(e)
			MyBase.DrawControlsArea(e)
			_LayoutControl.OnCustomDrawControlsArea(args)
			If args.Handled Then
				Return
			End If
			MyBase.DrawControlsArea(e)
		End Sub
	End Class
End Namespace
